server/middleware/setUpWebSecurity.ts

```diff
@@ -7,15 +8,35 @@ export default function setUpWebSecurity(): Router {
   // Secure code best practice - see:
   // 1. https://expressjs.com/en/advanced/best-practice-security.html,
   // 2. https://www.npmjs.com/package/helmet
+  router.use((_req: Request, res: Response, next: NextFunction) => {
+    res.locals.cspNonce = crypto.randomBytes(16).toString('hex')
+    next()
+  })
   router.use(
     helmet({
       contentSecurityPolicy: {
         directives: {
           defaultSrc: ["'self'"],
-          // Hash allows inline script pulled in from https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/template.njk
-          scriptSrc: ["'self'", 'code.jquery.com', "'sha256-+6WnXIl4mbFTCARd8N3COQmT3bJJmo32N8q8ZSQAIcU='"],
-          styleSrc: ["'self'", 'code.jquery.com'],
+          // This nonce allows us to use scripts with the use of the `cspNonce` local, e.g (in a Nunjucks template):
+          // <script nonce="{{ cspNonce }}">
+          // or
+          // <link href="http://example.com/" rel="stylesheet" nonce="{{ cspNonce }}">
+          // This ensures only scripts we trust are loaded, and not anything injected into the
+          // page by an attacker.
+          scriptSrc: [
+            "'self'",
+            'www.gstatic.com',
+            'code.jquery.com',
+            (_req: Request, res: Response) => `'nonce-${res.locals.cspNonce}'`,
+          ],
+          styleSrc: [
+            "'self'",
+            'www.gstatic.com',
+            'code.jquery.com',
```

Comment:
is jQuery domain needed if you've removed it in `layout.njk` and using it locally? Might be used by MoJ library I guess 🤷‍♂️

Reply:
I'll be honest, I'm not sure, I just pulled across from the template app. I'll check in a bit.

Reply:
@tpmcgowan I've removed jquery completely. I wonder if it's even needed in the template 🤔

Reply:
MoJ Pattern library says it needs jQuery (https://design-patterns.service.justice.gov.uk/get-started/setting-up-javascript/) but I guess that might be dependent on what components you're using. But the template has jQuery and jQuery UI in its `layout.njk` 🤔 .

Reply:
Good spot. We don't actually use any moj-frontend components in this app. I'll see about a PR to remove.